es74x/src/main/scala/tech/beshu/ror/es/request/context/types/SearchEsRequestContext.scala

```diff
@@ -40,14 +41,15 @@ class SearchEsRequestContext(actionRequest: SearchRequest,
 
   override protected def update(request: SearchRequest, indices: NonEmptyList[IndexName]): ModificationResult = {
     optionallyDisableCaching()
+    SearchQueryDecorator.applyFilterToQuery(request, threadPool)
     request.indices(indices.toList.map(_.value.value): _*)
     Modified
   }
 
   // Cache disabling for this request is crucial for document level security to work.
   // Otherwise we'd get an answer from the cache some times and would not be filtered
   private def optionallyDisableCaching(): Unit = {
-    if(esContext.involveFilters) {
+    if (esContext.involveFilters) {
```

Comment:
The whole objetive here is to leave this cache enabled. In order to do so, we need to add the filter to the search request from the acl rule.